### PR TITLE
fix issue: 2294, set tieredstore in AlluxioRuntime do not affect memory

### PR DESCRIPTION
**Pull request by hahchenchen**
Please make sure you have read and understood the contributing guidelines;
https://github.com/fluid-cloudnative/fluid/blob/master/CONTRIBUTING.md-->

### Ⅰ. Describe what this PR does
fix issue: #2294 , set tieredstore in AlluxioRuntime do not affect memory

### Ⅱ. Does this pull request fix one issue?

fix issue: #2294 

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]
The code changes may harm the consistency between different cache runtimes when handling the setting. I suggest to follow [JindoRuntime's logic](https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/jindofsx/transform.go#L497) when setting tieredstore's memory requests and limits.

File: pkg/ddc/alluxio/transform_resources.go [hunk 1]
```diff
@@ -60,39 +58,75 @@ func (e *AlluxioEngine) transformResourcesForWorker(runtime *datav1alpha1.Alluxi
...
 
-			memLimit.Add(req)
-
-			e.Log.Info("update the requirement for memory", "requirement", memLimit)
+			if runtime.Spec.Worker.Resources.Requests == nil ||
+				runtime.Spec.Worker.Resources.Requests.Memory() == nil ||
+				runtime.Spec.Worker.Resources.Requests.Memory().IsZero() ||
+				req.Cmp(*runtime.Spec.Worker.Resources.Requests.Memory()) > 0 {
+				needUpdated = true
+				needSetMem.Add(req)
```
Comment (allenhaozi) [inline pkg/ddc/alluxio/transform_resources.go:81]: Is req going to be null?

Comment (hahchenchen) [reply pkg/ddc/alluxio/transform_resources.go:81]: no, the req come from: `storageMap := tieredstore.GetLevelStorageMap(runtimeInfo)`

Comment (allenhaozi) [reply pkg/ddc/alluxio/transform_resources.go:81]: > no, the req come from: `storageMap := tieredstore.GetLevelStorageMap(runtimeInfo)`

got it, thank you

**Review by allenhaozi — APPROVED** [review]

**Review by TrafalgarZZZ — APPROVED** [review]
LGTM

**Review by cheyang — APPROVED** [review]

**Pull request merged**